src/main/java/service/FileReaderImpl.java

```diff
@@ -0,0 +1,65 @@
+package service;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import model.FruitTransaction;
+import util.ValidationDataException;
+
+public class FileReaderImpl implements FileReader {
+    private static final String SPLIT_REGEX = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> read(String inputFileName) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        try (BufferedReader reader = new BufferedReader(new java.io.FileReader(inputFileName))) {
+            reader.readLine();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                FruitTransaction transaction = extractTransaction(line);
+                transactions.add(transaction);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file: " + inputFileName);
+        }
+
+        return transactions;
+    }
+
+    private FruitTransaction extractTransaction(String line) {
```

Comment:
The logic is fine, but you should place it in a separate service. The reader should return raw content, like List of String, not a list of DTOs. Then you pass it to FruitTransactionParser (you may name it differently) and it returns a list of transactions. That's what we meant by the separation of concerns